kafka/producer.go

```diff
@@ -148,10 +151,21 @@ func (p *Producer) gethandle() *handle {
 	return &p.handle
 }
 
+// verifyProducer verifies if Producer can be used
+func (p *Producer) verifyProducer() bool {
+	if p == nil || p.isClosed {
+		return false
```

Comment:
This is not thread safe and I don't think it's worth making it thread safe. What's the use case for this? I understand it's useful to know if the producer is closed but I don't see a real benefit to this.

Reply:
@vivek-ng this is for cases where the Producer, Consumer or AdminClient have been closed, and are used after being closed, or a nil instance returned by New* is used without checking the err. It happens, and we want to give a meaningful error instead of a Go panic or a C segfault.

Reply:
@emasab that makes sense. Thanks for the context, this was super helpful 👍. Should we just use `if p == nil` for checking valid producer state (again, I'm not sure if this is thread safe either)? I'm concerned that checking `p.isClosed` is not thread safe  (same issue with consumer).

@PrasanthV454 I'd recommend adding detailed description to the PR explaining the reason for the change and what issue/problem this PR aims to solve (like what Emanuele mentioned in the comment) especially when you're contributing to open source projects used by many companies across the world. This will help people (maybe 2-3 years down the line) to debug issues faster when they look at PR and try to understand why this change was made. Please look at @milindl's PRs as examples on how to write good PR descriptions :)